x-pack/plugins/apm/server/routes/historical_data/has_historical_agent_data.ts

```diff
@@ -9,9 +9,15 @@ import { ProcessorEvent } from '@kbn/observability-plugin/common';
 import { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
 
 // Note: this logic is duplicated in tutorials/apm/envs/on_prem
-export async function hasHistoricalAgentData(apmEventClient: APMEventClient) {
+export async function hasHistoricalAgentData(
+  apmEventClient: APMEventClient,
+  { limitByDataTier } = { limitByDataTier: false }
```

Comment:
WDYT about making this required?

Reply:
I wonder if the data tier limiting should be an internal implementation detail, this function is used in three places IIRC.

Reply:
> WDYT about making this required?

Happy to do that

> I wonder if the data tier limiting should be an internal implementation detail, this function is used in three places IIRC.

How would that look?

Reply:
Two calls in the `hasHistoricalAgentData` similar to how you are using it in the endpoint.